x-pack/plugins/security_solution/cypress/integration/alerts_detection_rules_custom.spec.ts

```diff
@@ -124,78 +161,59 @@ describe('Detection rules, custom', () => {
     cy.get(RULES_TABLE).then(($table) => {
       cy.wrap($table.find(RULES_ROW).length).should('eql', 1);
     });
-    cy.get(RULE_NAME).invoke('text').should('eql', newRule.name);
-    cy.get(RISK_SCORE).invoke('text').should('eql', newRule.riskScore);
-    cy.get(SEVERITY).invoke('text').should('eql', newRule.severity);
-    cy.get('[data-test-subj="rule-switch"]').should('have.attr', 'aria-checked', 'true');
+    cy.get(RULE_NAME).should('have.text', newRule.name);
+    cy.get(RISK_SCORE).should('have.text', newRule.riskScore);
+    cy.get(SEVERITY).should('have.text', newRule.severity);
+    cy.get(RULE_SWITCH).should('have.attr', 'aria-checked', 'true');
 
     goToRuleDetails();
+    waitForTheRuleToBeExecuted();
```

Comment:
I would put this guard as close to the dependent assertions as possible: since (I think) it's only required for the alerts assertions, I would move this a bit further down.